File: app/src/main/java/com/fishweeb/practical/Vector2.java
package com.fishweeb.practical;

public class Vector2
{
    public float x;
    public float y;

    public Vector2()
    {
        x = 0;
        y = 0;
    }

    public Vector2(float x, float y)
    {
        this.x = x;
        this.y = y;
    }

    public void PlusEqual(Vector2 other)
    {
        x += other.x;
        y += other.y;
    }

    public void PlusEqual(Vector2 other,float _dt)
    {
        x += other.x * _dt;
        y += other.y * _dt;
    }

    public void MinusEqual(Vector2 other)
    {
        x -= other.x;
        y -= other.y;
    }

    public void MinusEqual(Vector2 other, float _dt)
    {
        x -= other.x * _dt;
        y -= other.y * _dt;
    }

    public Vector2 Plus(Vector2 other)
    {
        Vector2 tempVector = new Vector2(x,y);
        tempVector.PlusEqual(other);
        return tempVector;
    }

    public Vector2 Minus(Vector2 other)
    {
        Vector2 tempVector = new Vector2(x,y);
        tempVector.MinusEqual(other);
        return tempVector;
    }

    public Vector2 Normalized()
    {
        float distance = (float)Math.sqrt(x * x + y * y);
        return new Vector2(x / distance, y / distance);
    }

    public void Equals(Vector2 otherVector)
    {
        x = otherVector.x;
        y = otherVector.y;
    }

    public Vector2 Times(float num)
    {
        Vector2 tempVector = new Vector2(x,y);
        tempVector.x *= num;
        tempVector.y *= num;
        return tempVector;
    }

    public void TimesEquals(float num)
    {
        x *= num;
        y *= num;
    }

    public float DistanceSquaredFrom(Vector2 other)
    {
        float x = other.x - this.x;
        float y = other.y - this.y;
        float distSquared = x * x + y * y;
        return  distSquared;
    }
}

File: app/src/main/java/edu/androidgroupproject/Vector2.kt
package edu.androidgroupproject

import kotlin.math.sqrt

class Vector2 {
    var x: Float
    var y: Float

    constructor() {
        x = 0f
        y = 0f
    }

    constructor(x: Float, y: Float) {
        this.x = x
        this.y = y
    }

    fun PlusEqual(other: Vector2) {
        x += other.x
        y += other.y
    }

    fun PlusEqual(other: Vector2, _dt: Float) {
        x += other.x * _dt
        y += other.y * _dt
    }

    fun MinusEqual(other: Vector2) {
        x -= other.x
        y -= other.y
    }

    fun MinusEqual(other: Vector2, _dt: Float) {
        x -= other.x * _dt
        y -= other.y * _dt
    }

    fun Plus(other: Vector2): Vector2 {
        val tempVector = Vector2(x, y)
        tempVector.PlusEqual(other)
        return tempVector
    }

    fun Minus(other: Vector2): Vector2 {
        val tempVector = Vector2(x, y)
        tempVector.MinusEqual(other)
        return tempVector
    }

    fun Normalized(): Vector2 {
        val distance = sqrt((x * x + y * y).toDouble()).toFloat()
        return Vector2(x / distance, y / distance)
    }

    fun Equals(otherVector: Vector2) {
        x = otherVector.x
        y = otherVector.y
    }

    fun Times(num: Float): Vector2 {
        val tempVector = Vector2(x, y)
        tempVector.x *= num
        tempVector.y *= num
        return tempVector
    }

    fun TimesEquals(num: Float) {
        x *= num
        y *= num
    }

    fun DistanceSquaredFrom(other: Vector2): Float {
        val x = other.x - this.x
        val y = other.y - this.y
        val distSquared = x * x + y * y
        return distSquared
    }
}
